test/functional/lua/diagnostic_spec.lua

```diff
@@ -1104,7 +1104,7 @@ describe('vim.diagnostic', function()
 
     it('works with diagnostics before the start of the line', function()
       eq(
-        { 4, 0 },
+        { 4, -1 },
```

Comment:
This has changed because it now returns the original diagnostic that was set without clamping it's position.

Should it be changed? In general if I'm not mistaken it seems the preference is to keep diagnostics unmodified and handle invalid ranges gracefully only when displaying them, based on https://github.com/neovim/neovim/pull/15729